sql/catalyst/src/test/scala/org/apache/spark/sql/HashByteArrayBenchmark.scala

```diff
@@ -0,0 +1,145 @@
+    Hash byte arrays with length 31:    Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+    -------------------------------------------------------------------------------------------
+    Murmur3_x86_32                             41 /   41         51.7          19.3       1.0X
+    xxHash 64-bit                              33 /   34         63.7          15.7       1.2X
+    */
+    test(31, 42L, 1 << 10, 1 << 11)
+
+    /*
+    Running benchmark: Hash byte arrays with length 95
+      Running case: Murmur3_x86_32
+      Running case: xxHash 64-bit
+
+    Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz
+    Hash byte arrays with length 95:    Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+    -------------------------------------------------------------------------------------------
+    Murmur3_x86_32                             97 /   98         21.6          46.3       1.0X
+    xxHash 64-bit                              73 /   74         28.8          34.7       1.3X
+    */
+    test(64 + 31, 42L, 1 << 10, 1 << 11)
+
+    /*
+    Running benchmark: Hash byte arrays with length 287
+      Running case: Murmur3_x86_32
+      Running case: xxHash 64-bit
+
+    Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz
+    Hash byte arrays with length 287:   Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+    -------------------------------------------------------------------------------------------
+    Murmur3_x86_32                            282 /  284          7.4         134.7       1.0X
+    xxHash 64-bit                             113 /  114         18.5          54.0       2.5X
+    */
+    test(256 + 31, 42L, 1 << 10, 1 << 11)
+
+    /*
+    Running benchmark: Hash byte arrays with length 1055
+      Running case: Murmur3_x86_32
+      Running case: xxHash 64-bit
+
+    Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz
+    Hash byte arrays with length 1055:  Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+    -------------------------------------------------------------------------------------------
+    Murmur3_x86_32                           1050 / 1051          2.0         500.8       1.0X
+    xxHash 64-bit                             283 /  284          7.4         135.1       3.7X
+    */
+    test(1024 + 31, 42L, 1 << 10, 1 << 11)
+
+    /*
+    Running benchmark: Hash byte arrays with length 2079
+      Running case: Murmur3_x86_32
+      Running case: xxHash 64-bit
+
+    Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz
+    Hash byte arrays with length 2079:  Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+    -------------------------------------------------------------------------------------------
+    Murmur3_x86_32                           2065 / 2070          1.0         984.7       1.0X
+    xxHash 64-bit                             472 /  473          4.4         225.3       4.4X
+    */
+    test(2048 + 31, 42L, 1 << 10, 1 << 11)
+
+    /*
+    Running benchmark: Hash byte arrays with length 8127
```

Comment:
Copy & paste failure, should be: `Running benchmark: Hash byte arrays with length 8223`